pil-analyzer/src/statement_processor.rs

```diff
@@ -141,7 +130,19 @@ where
                 )
             }
             PilStatement::PublicDeclaration(source, name, polynomial, array_index, index) => {
-                self.handle_public_declaration(source, name, polynomial, array_index, index)
+                let (name, ty) = self.name_and_type_from_polynomial_name(
+                    PolynomialName {
+                        name,
+                        array_size: None,
+                    },
+                    Type::Expr,
+                )?;
```

Comment:
```suggestion
```
This doesn't really do anything here because we know it is not an array (array of publics that would be)